sherpa/astro/fake.py

```diff
@@ -82,8 +86,12 @@ def fake_pha(data, model,
         For all background ids not listed in this dictionary, the
         counts will be drawn from the PHA data of the background data set.
     id : str
-        String with id number if called from UI layer. This is only used for
-        certain error messages.
+        String with id number if called from UI layer. This is only
+        used for certain error messages.
+    method : callable or None
+        The routine used to simulate the data. If None (the default)
+        then sherpa.utils.poisson_noise is used, otherwise the function
+        must accept a single ndarray, returning a ndarray of the same shape.
```

Comment:
This is a clever addition that makes so much sense. Why did we not do that years ago?

Reply:
What are the choices for 'method'?

Reply:
Any callable (function or callable class) that you define yourself. Since we don't expect users to use this feature are lot, I don't think any more details are needed. There is a sensible default (None, which means Poisson) that should cover 98 % of all use cases.

Reply:
@anetasie - the docstring says

    method: callable or None

which is, I believe, the Python way to do it (without going into type hints which would be even less intelligible).